tests/test_coreg/test_affine.py

```diff
@@ -139,38 +244,49 @@ def test_coreg_example(self, verbose: bool = False) -> None:
         test_examples.py, but helps identify from where differences arise.
         """
 
+        # Use full DEMs here (to compare to original values from older package versions)
+        ref, tba = load_examples(crop=False)[0:2]
+        inlier_mask = ~self.outlines.create_mask(ref)
+
         # Run co-registration
         nuth_kaab = xdem.coreg.NuthKaab()
-        nuth_kaab.fit(self.ref, self.tba, inlier_mask=self.inlier_mask, verbose=verbose, random_state=42)
+        nuth_kaab.fit(ref, tba, inlier_mask=inlier_mask, verbose=verbose, random_state=42)
 
         # Check the output .metadata is always the same
-        shifts = (nuth_kaab.meta["shift_x"], nuth_kaab.meta["shift_y"], nuth_kaab.meta["shift_z"])
-        res = self.ref.res[0]
-        assert shifts == pytest.approx((-0.463 * res, -0.1339999 * res, -1.9922009))
+        shifts = (
+            nuth_kaab.meta["outputs"]["affine"]["shift_x"],
+            nuth_kaab.meta["outputs"]["affine"]["shift_y"],
+            nuth_kaab.meta["outputs"]["affine"]["shift_z"],
+        )
+        assert shifts == pytest.approx((-9.200801, -2.785496, -1.9818556))
```

Comment:
I see that the shifts change at the 2nd decimal (nearly 1st decimal when rounded). I'm curious if this could be related to the different bin size used previously and now (72 vs 80)?

Reply:
There's several possible reasons, in order of possible importance I would say:
1. We used to have forced filters in the Nuth and Kääb function (larger than a certain dh, I think 100 meters, and flat or very high slopes), those are now removed; To have those filters one can use the `dem_coregistration` workflow right now, and we should add the planned `CoregFilters` classes to add this directly in a pipeline soon,
2. Bin size as you mention,
3. We use to have a rounding + default tolerance parameters for the SciPy algorithm to make the result the same exactly on all OS... I might have to deal with this again once I get the CI running (now that GeoUtils PR is merged),
4. Interpolation is now very slightly different.

Reply:
It probably all comes from 1.